backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java

```diff
@@ -38,4 +58,51 @@ public void enrollVolunteer(String proposalId, Volunteer volunteer) {
         proposal.inscribeVolunteer(volunteer);
         proposalRepository.save(proposal);
     }
+
+    /**
+     * Fetch the ESAL of the database for get the ContactPerson, change status of the proposal and send an email with the revision of the reviser.
+     *
+     * @param proposalId : The id of the proposal to be revised.
+     * @param revisionDto : Contains the email reviser and the feedback if has it.
+     * @param proposalURI : URI the proposal to revise.
+     * @param accountId Account ID of logged user
+     */
+    public ProposalRevisionEmail requestChanges(
+            String proposalId,
+            ProposalRevisionDto revisionDto,
+            URI proposalURI,
+            String accountId) {
+
+        Proposal proposal = proposalRepository.fetch(proposalId);
+
+        if (ProposalStatus.REVIEW_PENDING.getId() != proposal.getStatus().getId()) {
+            throw new InvalidProposalStatusException();
+        }
+
+        ESAL esal = proposalRepository.fetch(proposalId).getEsal();
```

Comment:
You have previously fetched the proposal, there's no need to do it again.

Reply:
Ok, it's done.